probe-rs/src/architecture/arm/core/m0.rs

```diff
@@ -429,7 +434,7 @@ impl<'probe> CoreInterface for M0<'probe> {
 
         self.memory.write_word_32(Dhcsr::ADDRESS, value.into())?;
 
-        self.wait_for_core_halted()?;
+        self.wait_for_core_halted(Duration::from_millis(100))?;
```

Comment:
Why is this hardcoded, and not a parameter as well?

Reply:
I didn't want to clutter the interface too much and felt like whenever one wants to step and it takes too long to do that, timeouting would be fine. Other than with halt for example wehere you sometimes want to await a longer operation.

Do you disagree?